lib/api/core/models/repositories/profileRepository.js

```diff
@@ -167,38 +176,29 @@ class ProfileRepository extends Repository {
   }
 
   /**
-   * Populates a Profile object with the values contained in the data object.
-   *
    * @param {Profile} profile
-   * @param {object} data
-   * @returns {Promise<Profile>} Resolves to the hydrated Profile object.
+   * @returns {Promise<Profile>}
    */
-  hydrate (profile, data) {
-    if (data._source) {
-      const source = data._source;
-      delete data._index;
-      delete data._type;
-      delete data._version;
-      delete data.found;
-      delete data._source;
-      Object.assign(data, source);
-    }
+  initialize (profile) {
```

Comment:
I am not sure to agree. 

Comes with your proposal about turning current `initialize` methods into a per-repository factory.
The business factory is delegated to the parent `Repository` mainly because depending on its source (db or cache), the pojo object has a different structure.

I am afraid that adding a new intermediate pojo structure (or re-use an existing one), would just add some additional complexity.

About mutation: the naming of the method is quite clear to me that we expect it to modify the object given as parameter. 
As the mutation is idempotent, I don't see any harm in letting it that way.
Actually, making it returning a new structure is probably more confusing IMO:

```js
const p = new Profile();
const p2 = p.initialize();
p2.getRoles(); // ok
p.getRoles(); // throws
```
... or we would need to change the naming from `initialize` to something ugly like `getInitalizedProfileFrom`

Reply:
no, my suggestion is (with maybe a better method naming):
```js
const p = ProfileRepository.instantiate(data); // returns a `Profile` object "hydrated" with the data
const roles = p.getRoles(); // OK - never need to throw, as it is always initialized 
```